diff --git a/node_modules/oidc-provider/lib/models/client.js b/node_modules/oidc-provider/lib/models/client.js
index 903f909..6ef9bed 100644
--- a/node_modules/oidc-provider/lib/models/client.js
+++ b/node_modules/oidc-provider/lib/models/client.js
@@ -515,7 +515,18 @@ export default function getClient(provider) {
       const parsed = URLparse(value);
       if (!parsed) return false;
 
-      const match = this.redirectUris.find((allowed) => URLparse(allowed)?.href === parsed.href);
+      const match = this.redirectUris.find((allowed) => {
+        // Check if the allowed redirect ends with an asterisk
+        if (allowed.endsWith("*")) {
+          // Wildcard match
+          const base = allowed.slice(0, -1);
+          return parsed.href.startsWith(base);
+        } else {
+          // Exact match
+          return URLparse(allowed)?.href === parsed.href;
+        }
+      });
+
       if (
         !!match
         || this.applicationType !== 'native'
